src/main/java/software/amazon/smithy/lsp/diagnostics/DetachedDiagnostics.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.lsp.diagnostics;
+
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.DiagnosticSeverity;
+import org.eclipse.lsp4j.Range;
+import software.amazon.smithy.lsp.project.SmithyFile;
+import software.amazon.smithy.lsp.protocol.RangeAdapter;
+
+/**
+ * Diagnostics for when a Smithy file is not connected to a Smithy project via
+ * smithy-build.json or other build file.
+ */
+public final class DetachedDiagnostics {
```

Comment:
I expect to add more kinds of diagnostics, so I think they make sense here for now. Since the language server isn't really a library right now, it shouldn't matter too much, but no harm in adding it.